app/services/fees/estimate_pay_in_advance_service.rb

```diff
@@ -78,11 +78,11 @@ def subscriptions
       @subscriptions = subscriptions
         .where("date_trunc('second', started_at::timestamp) <= ?", timestamp)
         .where("terminated_at IS NULL OR date_trunc('second', terminated_at::timestamp) >= ?", timestamp)
-        .order(started_at: :desc)
+        .order('terminated_at DESC NULLS FIRST, started_at DESC')
     end
 
     def charges
-      @charges ||= event.subscription
+      @charges ||= subscriptions.first
```

Comment:
does it work for the case when there is no external_subscription_id and customer has multiple subscriptions?

Reply:
It's handle bye the `Events::ValidateCreationService` and it should raise an error